package/kedro_viz/models/flowchart.py

```diff
@@ -769,31 +725,22 @@ def set_run_command(cls, _):
             return f"kedro run --to-outputs={cls.data_node.name}"
         return None
 
-    @validator("plot", always=True)
-    def set_plot(cls, _):
-        if cls.data_node.is_plot_node():
-            return cls.data_node.kedro_obj.load()
-        return None
-
-    @validator("image", always=True)
-    def set_image(cls, _):
-        if cls.data_node.is_image_node():
-            return cls.data_node.kedro_obj.load()
-        return None
-
-    @validator("tracking_data", always=True)
-    def set_tracking_data(cls, _):
-        if cls.data_node.is_tracking_node():
-            return cls.data_node.kedro_obj.load()
-        return None
-
     @validator("preview", always=True)
     def set_preview(cls, _):
-        if not (cls.data_node.is_preview_node() and hasattr(cls.dataset, "_preview")):
+        if not (
+            not cls.data_node.disable_preview() and hasattr(cls.dataset, "preview")
+        ):
             return None
 
         try:
-            return cls.dataset._preview(**cls.data_node.get_preview_args())
+            preview_args = None
+            if cls.data_node.viz_metadata:
+                preview_args = cls.data_node.get_preview_args()
+            return (
+                cls.dataset.preview(**preview_args)
+                if preview_args is not None
+                else cls.dataset.preview()
+            )
```

Comment:
This is personal preference, but I find this code quite hard to read. I would personally change it to:

```python
preview_args = cls.data_node.get_preview_args() if cls.data_node.viz_metadata else None
if preview_args is not None:
    return cls.dataset.preview(**preview_args)
else:
    return cls.dataset.preview()

```